### DISCUSSION TITLE
Building for Sunton 4827S043R - screen flicker and touch support issues

### DISCUSSION THREAD

**Discussion by VauxhallViva** [HASwitchPlate/openHASP#648] [Hardware]
Hi, new here in this amazing project. Already setup a Sunton 2832S028R using the pre-compiled binary from the nightly builds with no issues. But I also have a 4827S043R with resistive touch, which is not available as an option in the nightly builds.

I followed the advice to use Gitpod to compile my own version. Was able to modify the file sunton-esp32-s3-tft.ini to add an entry for my device, changing the touch driver to "0x2046", but this would not compile. I then commented out the touch driver part and was able to build a binary and flash it. This worked, but of course no touch and the screen also suffered from flickering.

Found some posts (not related to openHASP) regarding the flickering, and by changing the build flag "TFT_DE" to zero, the screen flicker was gone.

But not able to get the touch to work yet, anyone else had any success with this and can share their solution?

**Comment by lucvan** [comment]
For anyone stuck here with **no touch on the 4827S043R**: the root cause is that openHASP's XPT2046 support is wired only into the TFT_eSPI and LovyanGFX display drivers, but this board uses **ArduinoGFX** for its RGB-parallel panel — so `TOUCH_DRIVER=0x2046` matches no dispatch branch in `touch_driver.h` and touch silently falls through to a no-op stub. (The display itself works; the flicker is separate — the porch / `TFT_PCLK_ACTIVE_NEG` timings in the board `.ini`.)

I've opened #1028, which adds a **standalone XPT2046 driver** that runs on its own SPI bus, independent of the display driver. On my 4827S043R the display + resistive touch both work, calibrated across the full 480×272 panel.

To try it before it's merged: build the `sunton-4827s043r_16MB` env from that branch (run `git submodule update --init lib/freetype` first). The touch pins were already in the board template (SCLK 12 / MISO 13 / MOSI 11 / CS 38 / IRQ 18); calibration is via `TOUCH_XPT2046_*` build flags — this panel needed `SWAP_XY=1`. Feedback on the PR welcome.
